frame/assets/src/functions.rs

```diff
@@ -178,15 +178,15 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		if amount.is_zero() {
 			return Success
 		}
-		let account = match Account::<T, I>::get(id, who) {
+		let account = match Account::<T, I>::get(&id, who) {
 			Some(a) => a,
 			None => return BalanceLow,
 		};
 		if account.status.is_frozen() {
 			return Frozen
 		}
 		if let Some(rest) = account.balance.checked_sub(&amount) {
-			if let Some(frozen) = T::Freezer::frozen_balance(id, who) {
+			if let Some(frozen) = T::Freezer::frozen_balance(id.clone(), who) {
```

Comment:
Can the first parameter take in a reference instead of making it clone now? Or do you think it's more of an API change and we should just leave this question for later?

Reply:
Yes we probably could, but since changes like that are often very far reaching I'd rather not do it in a single PR as that'll bloat it up even more, making it even harder to review and merge.